### PR TITLE
Report that FROM and JOIN do not support Subqueries

### PR DESCRIPTION
**Pull request by SimonHeimberg**
Make obvious that this is not supported.

### CODE REVIEW

**Comment by SimonHeimberg** [comment]
It is not obvious that Subqurries are not supported in FROM and JOIN. A question about how to do a subquery with dql pops up frequently.

Examples for questions:
 * http://stackoverflow.com/questions/27007090/inner-join-results-from-select-statement-using-doctrine-querybuilder#comment42611323_27007090
 * http://stackoverflow.com/questions/13163059/subquery-in-join-with-doctrine-dql
 * http://stackoverflow.com/questions/40617755/join-a-subquery-with-doctrine-dql/40715419#40715419


I can not run the unit test unfortunately because of too old php. Should I add a unit test? Where?

**Comment by Ocramius** [comment]
Requires test case. Can be added to https://github.com/doctrine/doctrine2/blob/6e6be3fdd92728b26a79576f7f91abe4080e54ff/tests/Doctrine/Tests/ORM/Query/QueryTest.php

If you can't run tests locally, travis-ci will do for you.

**Comment by SimonHeimberg** [comment]
Tests added

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 1]
```diff
@@ -242,4 +242,24 @@ public function testSetHydrationCacheProfileNull()
...
+     * @exptectedExcepionMessage Subquery
+     */
+    public function testSelectJoinSubquery()
+    {
+        $query = $this->_em->createQuery("select u from Doctrine\Tests\Models\CMS\CmsUser u JOIN (SELECT )");
+        $query->getResult();
+    }
+
+    /**
+     * @expectedException        Doctrine\ORM\Query\QueryException
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: Please use `$this->expectException(QueryException::class)` and `$this->exceptExceptionMessage('Subquery')` directly before the code that is supposed to trigger the failure (I suppose it's `$query->getResult()` here?

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 2]
```diff
@@ -242,4 +242,24 @@ public function testSetHydrationCacheProfileNull()
         $query->setHydrationCacheProfile(null);
         $this->assertNull($query->getHydrationCacheProfile());
     }
+
+    /**
+     * @expectedException        Doctrine\ORM\Query\QueryException
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: Please use `$this->expectException(QueryException::class)` and `$this->exceptExceptionMessage('Subquery')` directly before the code that is supposed to trigger the failure (I suppose it's `$query->getResult()` here?

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
@SimonHeimberg thanks for looking into the tests! This is indeed a good addition for 2.6 :-)

**Comment by hkdobrev** [comment]
@Ocramius @SimonHeimberg Is there anything else needed to be done for this? This would be a great addition!

**Comment by SimonHeimberg** [comment]
oops, tests failed, wait a minute...

**Comment by SimonHeimberg** [comment]
I finally managed to fix the typos, new commit ready for review now.

**Comment by Ocramius** [comment]
@SimonHeimberg this was manually rebased and merged into `master`, via doctrine/doctrine2@1e3bf6562e08b24b9973343b031e063ad6325d23, thanks! \o/

**Pull request closed without merging**